vendor/github.com/softlayer/softlayer-go/services/network.go

```diff
@@ -6301,7 +6301,7 @@ func (r Network_SecurityGroup) Offset(offset int) Network_SecurityGroup {
 }
 
 // Add new rules to a security group by sending in an array of template [[SoftLayer_Network_SecurityGroup_Rule (type)]] objects to be created.
-func (r Network_SecurityGroup) AddRules(ruleTemplates []datatypes.Network_SecurityGroup_Rule) (resp bool, err error) {
+func (r Network_SecurityGroup) AddRules(ruleTemplates []datatypes.Network_SecurityGroup_Rule) (resp string, err error) {
```

Comment:
With the latest API I see that the return result is `SoftLayer_Network_SecurityGroup_Request` per
http://sldn.softlayer.com/reference/services/softlayer_network_securitygroup/addrules.

Please test by vendoring latest softalyer-go.  You can generate the latest softayer-go by issuing `make generate` in the softlayer-go project. We should try with that updated library. I will raise issue in softlayer-go for this once we can confirm the latest API so obtained works fine. This should be used to validate the UAT.


```go

// Add new rules to a security group by sending in an array of template [[SoftLayer_Network_SecurityGroup_Rule (type)]] objects to be created.
func (r Network_SecurityGroup) AddRules(ruleTemplates []datatypes.Network_SecurityGroup_Rule) (resp datatypes.Network_SecurityGroup_Request, err error) {
	params := []interface{}{
		ruleTemplates,
	}
	err = r.Session.DoRequest("SoftLayer_Network_SecurityGroup", "addRules", params, &r.Options, &resp)
	return
}
```

Reply:
I think the api is going to change more. Right now it returns a request object that looks for example like `{"id":"98bb9c91fcdad1fa29d9a9d"}`. That id field does not seem to match any api to reuse. So I let it go as a string for now. We can change it later when it is more useful.